sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreProvider.scala

```diff
@@ -756,7 +760,7 @@ private[sql] class RocksDBStateStoreProvider
         case None =>
           // Create new store instance. The stamp should be defined
           // in this case
-          new RocksDBStateStore(version, stamp.get, readOnly)
+          new RocksDBStateStore(version, stamp.get, readOnly, forceSnapshotOnCommit)
```

Comment:
Lets also set `forceSnapshotOnCommit` for the case above, where we are reusing store. We reuse store for aggregation, if we don't set this, then we would never force snapshot for aggregation queries